common/flatpak-run.c

```diff
@@ -2961,8 +2961,40 @@ flatpak_run_app (const char     *app_ref,
 
   if (app_deploy != NULL)
     {
+      g_autofree const char **previous_ids = NULL;
+
       app_files = flatpak_deploy_get_files (app_deploy);
 
+      previous_ids = flatpak_deploy_data_get_previous_ids (app_deploy_data);
+      if (previous_ids)
+        {
+          for (i = 0; previous_ids[i] != NULL; i++)
+            {
+              g_autoptr (GFile) previous_data_dir = NULL;
+
```

Comment:
we should check at the start of the loop (each time around, because a previous iteration can create it) whether the current dir exists - if it does, we should break; and we have no more work to do

Reply:
hmm, currently this patch `break;` the loop as soon as we rename the previous_data_dir and symlinks it (successfully). So it does not makes sense to check the current dir at the start, maybe?